internal/nvcaps/nvcaps.go

```diff
@@ -36,27 +36,62 @@ const (
 	nvcapsDevicePath     = "/dev/nvidia-caps"
 )
 
-// MigMinor represents the minor number of a MIG device
-type MigMinor int
+// An Index represents a gpu, ci, or gi index.
+// We use uint32 as this typically maps to a device minor number.
+type Index uint32
 
-// MigCap represents the path to a MIG cap file
+// MigMinor represents the minor number of a MIG device
```

Comment:
Should we have consistency on the `// doc` I see for Index we have `// A VarName` but for the rest we use the regular `// VarName`